diff --git a/screen-4.9.1.orig/extern.h b/screen-4.9.1/extern.h
index d4e57df..e4790cd 100644
--- a/screen-4.9.1.orig/extern.h
+++ b/screen-4.9.1/extern.h
@@ -114,6 +114,7 @@ extern int   SetBaud __P((struct mode *, int, int));
 extern int   SttyMode __P((struct mode *, char *));
 extern int   CheckTtyname __P((char *));
 extern char  *GetPtsPathOrSymlink __P((int));
+extern char  *GetPtsCwd __P((int));
 
 /* mark.c */
 extern int   GetHistory __P((void));
diff --git a/screen-4.9.1.orig/process.c b/screen-4.9.1/process.c
index 14a7605..bad0c2d 100644
--- a/screen-4.9.1.orig/process.c
+++ b/screen-4.9.1/process.c
@@ -6176,6 +6176,10 @@ char *fn, **av;
 	case 'L':
 	  nwin.Lflag = 1;
 	  break;
+	case 'C':
+	  if (fore)
+	    nwin.dir = GetPtsCwd(fore->w_ptyfd);
+	  break;
 	default:
 	  Msg(0, "%s: screen: invalid option -%c.", fn, av[0][1]);
 	  break;
diff --git a/screen-4.9.1.orig/pty.c b/screen-4.9.1/pty.c
index 6791fd5..e3f2072 100644
--- a/screen-4.9.1.orig/pty.c
+++ b/screen-4.9.1/pty.c
@@ -423,3 +423,22 @@ char *GetPtsPathOrSymlink(int fd)
 
 	return tty_name;
 }
+
+char *GetPtsCwd(int fd)
+{
+
+	pid_t pgrp;
+	if ((pgrp = tcgetpgrp(fd)) == -1)
+		return NULL;
+
+	static char cwd[MAXPATHLEN];
+	ssize_t len;
+	/* len(/proc/PID/cwd) where PID max 64 bit int*/
+	char proccwd[6 + 20 + 4 + 1];
+	sprintf(proccwd, "/proc/%d/cwd", (int)pgrp);
+
+	if ((len = readlink(proccwd, cwd, MAXPATHLEN)) == -1)
+		return NULL;
+	cwd[len] = '\0';
+	return cwd;
+}
