--- screen.c.orig	2020-02-06 05:23:48.000000000 +0900
+++ screen.c	2020-03-26 19:03:38.675528800 +0900
@@ -1110,8 +1110,10 @@
           Panic(0, "Directory '%s' must be owned by root.", SockDir);
         n = (eff_uid == 0 && (real_uid || (st.st_mode & 0775) != 0775)) ? 0755 :
             (eff_gid == (int)st.st_gid && eff_gid != real_gid) ? 0775 : 0777;
+#ifndef __MSYS__
         if (((int)st.st_mode & 0777) != n)
           Panic(0, "Directory '%s' must have mode %03o.", SockDir, n);
+#endif
       }
       sprintf(SockPath, "%s/S-%s", SockDir, LoginName);
       if (access(SockPath, F_OK)) {
@@ -1146,8 +1148,10 @@
 #endif
   }
 
+#ifndef __MSYS__
   if ((st.st_mode & 0777) != 0700)
     Panic(0, "Directory %s must have mode 700.", SockPath);
+#endif
   if (SockMatch && index(SockMatch, '/'))
     Panic(0, "Bad session name '%s'", SockMatch);
   SockName = SockPath + strlen(SockPath) + 1;
